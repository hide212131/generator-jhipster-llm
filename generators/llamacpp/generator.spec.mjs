import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'llamacpp';
const SUB_GENERATOR_NAMESPACE = `jhipster-llm:${SUB_GENERATOR}`;

describe('SubGenerator llm of llm JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(SUB_GENERATOR_NAMESPACE)
        .withJHipsterConfig()
        .withOptions({
          ignoreNeedlesError: true,
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
