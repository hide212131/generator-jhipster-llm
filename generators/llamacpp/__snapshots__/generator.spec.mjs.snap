// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator llm of llm JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "models/README.md": {
    "stateCleared": "modified",
  },
  "models/download-model.mjs": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
}
`;
